SPECIFICATION.md

```diff
@@ -108,10 +108,15 @@ Interactions with the API may result in an error, these may be transport errors,
 Errors:
  
 - Must be wrapped in native errors or exceptions.
-- Should have a type that classifies the error
- 	* Transport error (could not reach the API).
- 	* API Errors (error response from API).
- 	* Validation Errors (unexpected data from API).
+- Should have a type that classifies the error:
+    * Transport Errors (could not reach the API, did not receive a response from the API).
+    * Request Errors (invalid data, bad credentials, API rejected due to invalid request, 400 class errors).
```

Comment:
Yep, ClientError for 4xx and ServerError for 5xx works well--matches the HTTP language, makes it clear where the fault lies, and easy to make more specific e.g. AuthenticationError is a ClientError.